queries/runescript/highlights.scm

```diff
@@ -0,0 +1,100 @@
+; Variables
+[
+  (identifier)
+  (local_variable)
+  (game_variable)
+  (constant_variable)
+  (parameter_variable)
+  (declaration_variable)
+] @variable
+
+; Scripts
+(script
+  trigger: (identifier) @keyword.directive
+  subject: (identifier) @function)
+
+; Calls
+(call
+  [
+    "~"
+    "@"
+  ]? @function
+  callee: (identifier) @function)
+
+; Literals
+((literal) @boolean
+  (#any-of? @boolean "true" "false"))
+
+(literal
+  "null" @constant.builtin)
+
+(integer_literal) @number
+
+(coord_literal) @string.special
+
+(comment) @comment @spell
+
+(string_fragment) @string
+
+(string_interpolation
+  [
+    "<"
+    ">"
+  ] @punctuation.special)
+
+(string_tag) @punctuation.special
+
+; Types
+(type) @type
+
+(parameter_type) @type
+
+; Tokens
+[
+  ";"
+  ":"
+  ","
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "{"
+  "}"
+] @punctuation.bracket
```

Comment:
Also square brackets

Reply:
Square brackets are only used to define the script type (header), nowhere else. I don't think that the should be counted as normal punctuation here. (Feel free to correct me if you think I'm wrong).

Reply:
Hmm I think it would still make sense to highlight them like this because they are still brackets, but if you prefer I could see maybe an argument for `@punctuation.special` maybe? cc @clason

Reply:
I'm OK with going with `@punctuation.special`